src/core/adapters/UniswapV2PoolForkAdapter.ts

```diff
@@ -238,11 +232,32 @@ export abstract class UniswapV2PoolForkAdapter
       token1Address: string
     }[]
   > {
+    // Volume and reserve filters have been added to avoid pairs that are not useful
     const response = await fetch(subgraphUrl, {
       method: 'POST',
       headers: { 'Content-Type': 'application/json' },
       body: JSON.stringify({
-        query: `{ pairs(first: ${this.MAX_FACTORY_PAIRS} where: {volumeUSD_gt: ${this.MIN_SUBGRAPH_VOLUME}} orderBy: reserveUSD orderDirection: desc) {id token0 {id} token1 {id}}}`,
+        query: `
+          {
+            pairs(
+              first: ${this.MAX_FACTORY_PAIRS}
+              where: {
+                volumeUSD_gt: ${this.MIN_SUBGRAPH_VOLUME}
+                reserve0_gte: ${this.MIN_TOKEN_RESERVE}
+                reserve1_gte: ${this.MIN_TOKEN_RESERVE}
```

Comment:
I added this MIN_TOKEN_RESERVE filter (set to 1) because, otherwise, there were a lot of completely useless pools at top position with only 2 holders. I'm unsure as to why those pools exists, but this filters them out.